### PR TITLE
Update documentation for cloud-init

### PR DESCRIPTION
**Pull request by 33percent**
Updated the cloud-init docs as referenced in the source document and provided source link too.

https://github.com/33percent/terraform-oci-oke/blob/issue-537/docs/cloudinit.adoc#configuring-cloud-init-for-the-nodepools

### CODE REVIEW

**Review by markxnelson — DISMISSED** [review]

**Review by hyder — CHANGES_REQUESTED** [review]
Hi. sorry for the delay. Can you please do the following:

1. Explicitly split the cloud-init doc into common, nodepool specific with headings etc.
2. For nodepool specific, can you please verify with @karthicgit how this mechanism work and add it?
3. Remove the section from https://github.com/oracle-terraform-modules/terraform-oci-oke/blob/main/docs/instructions.adoc#configuring-cloud-init-for-the-nodepools
4. Add a link to the cloud-init to the configuration docs and main docs

**Comment by hyder** [comment]
Closes #537

**Comment by 33percent** [comment]
Hi @hyder , Can you please review.

1. Explicitly split the cloud-init doc into common, nodepool specific with headings etc.
**Split the cloud init based on common script and node pool specific**

2. For nodepool specific, can you please verify with @karthicgit how this mechanism work and add it?
**Clarified with him. Made the changes accordingly @karthicgit , Can you review this once**

3. Remove the section from https://github.com/oracle-terraform-modules/terraform-oci-oke/blob/main/docs/instructions.adoc#configuring-cloud-init-for-the-nodepools
**Removed.**

4. Add a link to the cloud-init to the configuration docs and main docs
**Added**

File: docs/cloudinit.adoc [hunk 1]
```diff
@@ -6,33 +6,21 @@
 :toc: auto
 
 :uri-cloudinit: https://docs.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengusingcustomcloudinitscripts.htm
+:uri-source-cloudinit-doc: https://github.com/oracle-terraform-modules/terraform-oci-oke/blob/main/docs/instructions.adoc#configuring-cloud-init-for-the-nodepools
 
-== Instructions:
+== Configuring Cloud init scripts for Nodepools:
```
Comment (hyder) [inline docs/cloudinit.adoc:11]: It's cloud-init

File: docs/cloudinit.adoc [hunk 2]
```diff
@@ -6,33 +6,21 @@
...
+Cloud init scripts passed to the node pools can be passed in 2 different ways.
```
Comment (hyder) [inline docs/cloudinit.adoc:12]: cloud-init

File: docs/cloudinit.adoc [hunk 3]
```diff
@@ -6,33 +6,21 @@
...
-* The default cloud-init script used by this module is as below:
-----
-#!/bin/bash
-
-# DO NOT MODIFY
-curl --fail -H "Authorization: Bearer Oracle" -L0 http://169.254.169.254/opc/v2/instance/metadata/oke_init_script | base64 --decode >/var/run/oke-init.sh
-
-## run oke provisioning script
-bash -x /var/run/oke-init.sh
+* *Using `cloudinit_nodepool_common`*: When a script is passed to the cloudinit_nodepool_common, this init doc is executed in all the node pools.
```
Comment (hyder) [inline docs/cloudinit.adoc:14]: Elaborate on each under their own sections e.g. how to customize the common script

File: docs/cloudinit.adoc [hunk 4]
```diff
@@ -6,33 +6,21 @@
...
 
-### adjust block volume size
-/usr/libexec/oci-growfs -y
-
-timedatectl set-timezone ${worker_timezone}
+* *Using `cloudinit_nodepool`*: When custom scripts for each node pool is required to be executed, _cloudinit_nodepool_ can be used. It takes a map of values where each key is the node pool name.
+Example:
```
Comment (hyder) [inline docs/cloudinit.adoc:17]: Elaborate on each under their own sections e.g. how to customize the node pool specific script

File: docs/cloudinit.adoc [hunk 5]
```diff
@@ -6,33 +6,21 @@
...
-
-* To use specific cloud-init script for a nodepool pass the script as input variable to `cloudinit_nodepool` as a map.This will take precedence over `cloudinit_nodepool_common` for that nodepool.
-Ex: cloudinit_nodepool        = {
-  #np1 = "/tmp/np1cloudinit.sh"
-  #np3 = "/tmp/np3cloudinit.sh"
-#}
-
-
-
+The precedence of execution is if a custom script for the node pool is specified (`cloudinit_nodepool`), it will be executed. If it is not provided, then the `cloudinit_nodepool_common` script will be executed. Else the default `worker.template.sh` will be executed incase both are not provided.
```
Comment (hyder) [inline docs/cloudinit.adoc:26]: This should go to the top section then

**Review by hyder — CHANGES_REQUESTED** [review]
Please split as suggested

File: docs/cloudinit.adoc (hunk 1 above)
Comment (33percent) [reply docs/cloudinit.adoc]: This is changed.

File: docs/cloudinit.adoc (hunk 2 above)
Comment (33percent) [reply docs/cloudinit.adoc]: This is changed.

File: docs/cloudinit.adoc (hunk 3 above)
Comment (33percent) [reply docs/cloudinit.adoc]: Added examples for both types.

File: docs/cloudinit.adoc (hunk 4 above)
Comment (33percent) [reply docs/cloudinit.adoc]: Added examples for both types.

File: docs/cloudinit.adoc (hunk 5 above)
Comment (33percent) [reply docs/cloudinit.adoc]: This is moved to the top of the section.

**Comment by 33percent** [comment]
Hi, @hyder ,
The requested changes are updated. Can you please review!?

**Pull request closed without merging**